controllers/kubeadmbootstrapconfig_controller.go

```diff
@@ -56,3 +87,31 @@ func (r *KubeadmBootstrapConfigReconciler) SetupWithManager(mgr ctrl.Manager) er
 		For(&kubeadmv1alpha1.KubeadmBootstrapConfig{}).
 		Complete(r)
 }
+func (r *KubeadmBootstrapConfigReconciler) Map(object handler.MapObject) []reconcile.Request {
+	log := r.Log.WithValues("ns", object.Meta.GetNamespace(), "name", object.Meta.GetName())
+
+	log.Info("mapping machine to config")
+
+	result := []reconcile.Request{}
+	m := &clusterapiv1alpha2.Machine{}
+	key := client.ObjectKey{Namespace: object.Meta.GetNamespace(), Name: object.Meta.GetName()}
+	if err := r.Client.Get(context.Background(), key, m); err != nil {
+		return result
+	}
+
+	// There is no config reference for this machine, so don't enqueue anything
+	if m.Spec.Bootstrap.ConfigRef == nil {
+		return result
+	}
+
+	// Enqueue the associated config ref (not sure if this is actually what this should do?)
+	if m.Spec.Bootstrap.ConfigRef.Name != "" && m.Spec.Bootstrap.ConfigRef.Namespace != "" {
```

Comment:
Should we check if the APIGroup and Kind actually matches the object we're reconciling?

Reply:
Example: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/896/files#diff-fcb771956a4642d026c25a8ef9815e0bR155

Reply:
Yes, I'm still working on this code so I'll add it in a following PR that will actually do something interesting.